### PR TITLE
[test] Fix TestChangelogConsumer and address related issues

### PR DESCRIPTION
**Pull request by sushantmane**
## Fix TestChangelogConsumer and address related issues

- Fixed regression in `TestChangelogConsumer` caused by PR #1328, where it attempted to
register incompatible schemas.  
- Removed the creation of a pub-sub broker in the test as it was incorrectly relying on  
the ZK state of child controllers' Kafka broker.  
- Miscellaneous: Ensured proper cleanup by closing the cluster setup during teardown in
`TestDumpIngestionContext`.

## How was this PR tested?

CI

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/consumer/TestChangelogConsumer.java [hunk 1]
```diff
@@ -687,89 +674,82 @@ public void testSpecificRecordBootstrappingVeniceChangelogConsumer() throws Exce
...
-        Assert.assertEquals(polledChangeEventsList.size(), 2);
-      });
-      Assert.assertTrue(
-          polledChangeEventsMap.get(Integer.toString(10000)).getValue().getCurrentValue() instanceof SpecificRecord);
+    PubSubBrokerWrapper localKafka = multiRegionMultiClusterWrapper.getChildRegions().get(0).getPubSubBrokerWrapper();
+    Properties consumerProperties = new Properties();
+    String localKafkaUrl = localKafka.getAddress();
+    consumerProperties.put(KAFKA_BOOTSTRAP_SERVERS, localKafkaUrl);
+    consumerProperties.put(CLUSTER_NAME, clusterName);
+    consumerProperties.put(ZOOKEEPER_ADDRESS, localZkServer.getAddress());
```
Comment (nisargthakkar) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/consumer/TestChangelogConsumer.java:683]: Can you see if we can remove this? No client should depend on ZK directly

Comment (nisargthakkar) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/consumer/TestChangelogConsumer.java:683]: There's one more place down in the next test that also has the same

File: internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/consumer/TestChangelogConsumer.java
Comment (nisargthakkar) [inline internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/consumer/TestChangelogConsumer.java:1]: Looks like CC is not happy with the missing Zk address. Seems like a future incident to me, TBH. But we don't need to solve it in this PR

Comment (sushantmane) [reply internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/consumer/TestChangelogConsumer.java:1]: Maybe. I'm not gonna fix it in this PR.  

cc: @ZacAttack @sixpluszero

**Review by nisargthakkar — APPROVED** [review]

**Pull request merged**